### PR TITLE
Expose bpf-lb-sock-hostns-only in cilium status

### LINKED ISSUE #24160: Expose `bpf-lb-sock-hostns-only` in cilium status
Here is the description for the flag - `Skip socket LB for services when inside a pod namespace, in favor of service LB at the pod interface. Socket LB is still used when in the host namespace. Required by service mesh (e.g., Istio, Linkerd)`. 

This needs to be exposed as part of the `KubeProxyReplacement` section in `cilium status --verbose`:

```
KubeProxyReplacement Details:
  Status:                 Disabled
  Socket LB:              Disabled
  Socket LB Tracing:      Disabled
  Session Affinity:       Disabled
  Graceful Termination:   Enabled
  NAT46/64 Support:       Disabled
  Services:
  - ClusterIP:      Enabled
  - NodePort:       Disabled
  - LoadBalancer:   Disabled
  - externalIPs:    Disabled
  - HostPort:       Disabled
```

Here is the link to the related code - https://github.com/cilium/cilium/blob/master/pkg/client/client.go#L631.

### PR DESCRIPTION
**Pull request by romanspb80**
BPFSocketLBHostnsOnly is outputted in the Kube Proxy Replacement section in cilium status --verbose:

$ cilium status --verbose
[...]
KubeProxyReplacement Details:
  [...]
  BPFSocketLBHostnsOnly:   Enabled
  [...]

Fixes: #24160

Please ensure your pull request adheres to the following guidelines:

- [ ] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [ ] All code is covered by unit and/or runtime tests where feasible.
- [ ] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [ ] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [ ] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [ ] Provide a title or release-note blurb suitable for the release notes.
- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/master/USERS.md)
- [ ] Thanks for contributing!

Fixes: #issue-number

```release-note
<!-- Enter the release note text here if needed or remove this section! -->
```

### CODE REVIEW

**Review by aanm — APPROVED** [review]

**Comment by aanm** [comment]
It seems the CI is failing due the changes made into this PR https://github.com/cilium/cilium/actions/runs/4526983771/jobs/7972485978?pr=24570

File: api/v1/models/kube_proxy_replacement.go [hunk 1]
```diff
@@ -43,6 +43,9 @@ type KubeProxyReplacement struct {
 	// mode
 	// Enum: [Disabled Strict Probe Partial]
 	Mode string `json:"mode,omitempty"`
+
+	// flag bpf-lb-sock-hostns-only
+	BPFSocketLBHostnsOnly bool `json:"bpfSocketLBHostnsOnly,omitempty"`
```
Comment (aditighag) [inline api/v1/models/kube_proxy_replacement.go]: This should go under `KubeProxyReplacementFeatures` (see above L 41). Please take a look at how other flags are exposed.

**Review by aditighag — CHANGES_REQUESTED** [review]

File: pkg/client/client.go [hunk 2]
```diff
@@ -633,6 +638,7 @@ func FormatStatusResponse(w io.Writer, sr *models.StatusResponse, sd StatusDetai
 		fmt.Fprintf(tab, "  Status:\t%s\n", sr.KubeProxyReplacement.Mode)
 		fmt.Fprintf(tab, "  Socket LB:\t%s\n", socketLB)
 		fmt.Fprintf(tab, "  Socket LB Tracing:\t%s\n", socketLBTracing)
+		fmt.Fprintf(tab, "  BPFSocketLBHostnsOnly:\t%s\n", bpfSocketLBHostnsOnly)
```
Comment (borkmann) [inline pkg/client/client.go]: nit, this should better say:

`Socket LB Coverage:` and then spill out either `Hostns-only` or `Full` for example.

Comment (romanspb80) [reply pkg/client/client.go]: Ok. Done.

File: api/v1/models/kube_proxy_replacement.go (hunk 1 above)
Comment (romanspb80) [reply api/v1/models/kube_proxy_replacement.go]: Thanks. I've added the flag to the openapi

**Review by borkmann — APPROVED** [review]

File: pkg/client/client.go (hunk 2 above)
Comment (borkmann) [reply pkg/client/client.go]: Looks great, thanks @romanspb80 . Could you still add your Signed-off-by to the commit message? If you look intothe git log, you'll see that all commits have it. Then we should be good to move forward and merge. Thanks!

**Review by aditighag — APPROVED** [review]
Thanks for your contribution. LGTM... This is ready for merge once you add "Signed-off-by:" in the commit message.

File: api/v1/models/kube_proxy_replacement.go (hunk 1 above)
Comment (aditighag) [reply api/v1/models/kube_proxy_replacement.go]: This is still not addressed. You've added the flag under `KubeProxyReplacement`, and not `KubeProxyReplacementFeatures`. Was there an issue in adding it under the latter?

**Review by aditighag — CHANGES_REQUESTED** [review]
https://github.com/cilium/cilium/pull/24570#discussion_r1153562110

**Comment by romanspb80** [comment]
> [#24570 (comment)](https://github.com/cilium/cilium/pull/24570#discussion_r1153562110)

I missed it (((.
Done. Pls, check

**Review by aditighag — APPROVED** [review]
Thanks!
Integration test failure is same as - https://github.com/cilium/cilium/issues/24653. 
The PR is ready to merge.

**Pull request merged**